### PR TITLE
Add terraform import resource

### PR DESCRIPTION
**Pull request by shiyuhang0**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

1. Add terraform import resource
2. optimize other terraform docs

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/terraform-use-import-resource.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+- Create import tasks in serverless tier and dedicated tier.
+- LOCAL and S3 import type.
+- Cancel import tasks.
+
+You can learn how to manage an import task with the `tidbcloud_import` resource in this document.
+
+## Before you start
+
+- Learn how to [Get TiDB Cloud Terraform Provider](/tidb-cloud/terraform-get-tidbcloud-provider.md) first.
+- A serverless tier or dedicated tier is required.
```
Comment (zhangyangyu) [inline tidb-cloud/terraform-use-import-resource.md:19]:
```suggestion
- A serverless tier or dedicated tier cluster is required.
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    terraform {
+     required_providers {
+       tidbcloud = {
+         source = "tidbcloud/tidbcloud"
+         version = "~> 0.2.0"
+       }
+     }
+   }
+
+   provider "tidbcloud" {
```
Comment (zhangyangyu) [inline tidb-cloud/terraform-use-import-resource.md:47]: weird indent

File: tidb-cloud/terraform-use-import-resource.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+           table  = "import_test"
+       }
+       total_files                   = 0
+       total_size                    = "31"
+       total_tables_count            = 1
+       type                          = "LOCAL"
+   }
+   ```
+
+5. Use `terraform refersh` to update the status after several minutes
```
Comment (zhangyangyu) [inline tidb-cloud/terraform-use-import-resource.md:128]:
```suggestion
5. Use `terraform refresh` to update the status after several minutes
```

**Review by zhangyangyu — APPROVED** [review]

File: tidb-cloud/terraform-use-import-resource.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+tidbcloud_import.example_local: Destroying... [id=781074]
+╷
+│ Error: Delete Error
+│ 
+│ Unable to call CancelImport, got error: [DELETE /api/internal/projects/{project_id}/clusters/{cluster_id}/imports/{id}][500] CancelImport default  &{Code:59900104 Details:[] Message:failed to cancel
+│ import}
+╵
+```
+
+Cancel an import task which status is `IMPORTING`:
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:215]:
```suggestion
Cancel an import task whose status is `IMPORTING`:
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+  
+tidbcloud_import.example_local: Destroying... [id=781074]
+tidbcloud_import.example_local: Destruction complete after 0s
+
+Destroy complete! Resources: 1 destroyed.  
+```
+
+## Configurations
+
+See [configuration documentation](https://registry.terraform.io/providers/tidbcloud/tidbcloud/latest/docs/resources/import) to get all the available configurations for import resource.
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:236]:
```suggestion
See [configuration documentation](https://registry.terraform.io/providers/tidbcloud/tidbcloud/latest/docs/resources/import) to get all the available configurations for the import resource.
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+      }
+      target_table = {
+        database = "test"
+        table  = "import_test"
+      }
+      file_name = "your_csv_path"
+    }
+    ```
+
+   Replace resource values (such as project ID and cluster ID) in the file with your own. And you can find the details of `csv_format` in [CSV Configurations for Importing Data](/tidb-cloud/csv-config-for-import-data.md).
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:68]: https://docs.pingcap.com/tidbcloud/csv-config-for-import-data 这个文档里都是 UI 上的配置项名称，对应到这里的 .tf 配置文件，是否可以给一个配置项名称列表呢，类似 https://docs.pingcap.com/tidbcloud/ticloud-import-start-local#inherited-flags 这里的

File: tidb-cloud/terraform-use-import-resource.md [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+   |    2 | Bob   |   30 |
+   +------+-------+------+
+   2 rows in set (0.24 sec)
+   ```
+
+### Update
+
+Import tasks cannot be updated.
+
+### Delete
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:189]:
```suggestion
### Delete an import task
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+   +------+-------+------+
+   | id   | name  | age  |
+   +------+-------+------+
+   |    1 | Alice |   20 |
+   |    2 | Bob   |   30 |
+   +------+-------+------+
+   2 rows in set (0.24 sec)
+   ```
+
+### Update
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:185]:
```suggestion
### Update an import task
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+## Before you start
+
+- Learn how to [Get TiDB Cloud Terraform Provider](/tidb-cloud/terraform-get-tidbcloud-provider.md) first.
+- A serverless tier or dedicated tier cluster is required.
+
+## Manage the import resource
+
+Here is an example on how to manage a LOCAL import task with import resource.
+
+### Create
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:25]:
```suggestion
### Create and run an import task
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+You can learn how to manage an import task with the `tidbcloud_import` resource in this document.
+
+## Before you start
+
+- Learn how to [Get TiDB Cloud Terraform Provider](/tidb-cloud/terraform-get-tidbcloud-provider.md) first.
+- A serverless tier or dedicated tier cluster is required.
+
+## Manage the import resource
+
+Here is an example on how to manage a LOCAL import task with import resource.
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:23]:
```suggestion
You can manage either a local import task or an Amazon S3 import task using the import resource. The following sections take a local import task as an example to show you how to manage the import resource.
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+        database = "test"
+        table  = "import_test"
+      }
+      file_name = "your_csv_path"
+    }
+    ```
+
+   Replace resource values (such as project ID and cluster ID) in the file with your own. And you can find the details of `csv_format` in [CSV Configurations for Importing Data](/tidb-cloud/csv-config-for-import-data.md).
+
+3. Run the `terraform apply` command and type `yes` to confirm the creation:
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:70]:
```suggestion
3. Run the `terraform apply` command to create the import task, and then type `yes` to confirm the creation and start the import:
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+   2 rows in set (0.24 sec)
+   ```
+
+### Update
+
+Import tasks cannot be updated.
+
+### Delete
+
+Delete is actually cancel in import resource, and you can not cancel a `COMPLETED` import task.
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:191]:
```suggestion
For Terraform, deleting an import task means canceling the corresponding import resource. 
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 13]
```diff
@@ -0,0 +1,236 @@
...
+
+Cancel an import task which has completed:
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:193]:
```suggestion
you cannot cancel a `COMPLETED` import task. For example, if you cancel an import task that is completed, you will see get the `Delete Error` as follows:
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 14]
```diff
@@ -0,0 +1,236 @@
...
+
+Import resource supports:
+
+- Create import tasks in serverless tier and dedicated tier.
+- LOCAL and S3 import type.
+- Cancel import tasks.
+
+You can learn how to manage an import task with the `tidbcloud_import` resource in this document.
+
+## Before you start
```
[the remark this answers is not in the document]
Comment (shiyuhang0) [reply tidb-cloud/terraform-use-import-resource.md:16]: Done

File: tidb-cloud/terraform-use-import-resource.md (hunk 6 above)
Comment (shiyuhang0) [reply tidb-cloud/terraform-use-import-resource.md]: Done

**Review by zhangyangyu — APPROVED** [review]

File: tidb-cloud/terraform-use-import-resource.md [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+       total_files                   = 0
+       total_size                    = "31"
+       total_tables_count            = 1
+       type                          = "LOCAL"
+   }
+   ```
+
+   When the status turns to `COMPLETED`, it indicates that the import task is finished.
+
+6. Check the import data with MySQL CLI:
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:171]: Do we need to mention "MySQL CLI" or MySQL clients in the prerequisites?

Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:171]:
```suggestion
6. Check the imported data with MySQL CLI:
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 16]
```diff
@@ -0,0 +1,235 @@
...
+tidbcloud_import.example_local: Destroying... [id=781074]
+╷
+│ Error: Delete Error
+│ 
+│ Unable to call CancelImport, got error: [DELETE /api/internal/projects/{project_id}/clusters/{cluster_id}/imports/{id}][500] CancelImport default  &{Code:59900104 Details:[] Message:failed to cancel
+│ import}
+╵
+```
+
+Cancel an import task whose status is `IMPORTING`:
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:214]:
```suggestion
You can cancel an import task whose status is `IMPORTING`. For example:
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 17]
```diff
@@ -0,0 +1,235 @@
+---
+title: Use Import Resource
+summary: Learn how to manage the import task using the import resource.
+---
+
+# Use Import Resource
+
+Import resource supports:
+
+- Create import tasks in serverless tier and dedicated tier.
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:10]:
```suggestion
- Create import tasks for Serverless Tier and Dedicated Tier clusters.
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 18]
```diff
@@ -0,0 +1,235 @@
+---
+title: Use Import Resource
+summary: Learn how to manage the import task using the import resource.
+---
+
+# Use Import Resource
+
+Import resource supports:
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:8]:
```suggestion
You can learn how to import data to a TiDB Cloud cluster with the `tidbcloud_import` resource in this document.

The features of the `tidbcloud_import` resource include the following:
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 19]
```diff
@@ -0,0 +1,235 @@
...
+
+- Create import tasks in serverless tier and dedicated tier.
+- LOCAL and S3 import type.
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:11]:
```suggestion
- Support importing data from local disks or from Amazon S3 buckets.
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 20]
```diff
@@ -0,0 +1,235 @@
...
+- Cancel import tasks.
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:12]:
```suggestion
- Support canceling ongoing import tasks.
```

File: tidb-cloud/terraform-use-import-resource.md (hunk 18 above)
Comment (qiancai) [reply tidb-cloud/terraform-use-import-resource.md:8]: change reason: This is to use a similar introductory sentence as other Terraform integration docs.

File: tidb-cloud/terraform-use-import-resource.md [hunk 21]
```diff
@@ -0,0 +1,235 @@
...
+  
+tidbcloud_import.example_local: Destroying... [id=781074]
+tidbcloud_import.example_local: Destruction complete after 0s
+
+Destroy complete! Resources: 1 destroyed.  
+```
+
+## Configurations
+
+See [configuration documentation](https://registry.terraform.io/providers/tidbcloud/tidbcloud/latest/docs/resources/import) to get all the available configurations for the import resource.
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:235]: https://registry.terraform.io/providers/tidbcloud/tidbcloud/latest/docs/resources/import returns 404 Not Found.

File: tidb-cloud/terraform-use-import-resource.md [hunk 22]
```diff
@@ -0,0 +1,235 @@
...
+You can learn how to manage an import task with the `tidbcloud_import` resource in this document.
+
+## Prerequisites
+
+- [Get TiDB Cloud Terraform Provider](/tidb-cloud/terraform-get-tidbcloud-provider.md).
+- [Create a Serverless Tier or Dedicated Tier cluster](/tidb-cloud/create-tidb-cluster.md).
+
+## Manage the import resource
+
+You can manage either a local import task or an Amazon S3 import task using the import resource. The following sections take a local import task as an example to show you how to manage the import resource.
```
[the remark this answers is not in the document]
Comment (shiyuhang0) [reply tidb-cloud/terraform-use-import-resource.md:23]: Not supported yet. it depends on the ability of API which only support s3 and local now
image

File: tidb-cloud/terraform-use-import-resource.md (hunk 21 above)
Comment (shiyuhang0) [reply tidb-cloud/terraform-use-import-resource.md:237]: successfully access for me. Maybe the problem of network?
image

File: tidb-cloud/terraform-use-import-resource.md (hunk 15 above)
Comment (shiyuhang0) [reply tidb-cloud/terraform-use-import-resource.md]: I think we needn't. It is not a necessary step for import, just check the results for import

File: tidb-cloud/terraform-use-cluster-resource.md [hunk 23]
```diff
@@ -5,10 +5,16 @@ summary: Learn how to use the cluster resource to create and modify a TiDB Cloud
...
 
-You can learn how to create and modify a TiDB Cloud cluster with the `tidbcloud_cluster` resource in this document.
+You can learn how to manage a TiDB Cloud cluster with the `tidbcloud_cluster` resource in this document.
 
 In addition, you will also learn how to get the necessary information with the `tidbcloud_projects` and `tidbcloud_cluster_specs` data sources.
 
+The features of the `tidbcloud_cluster` resource include the following:
+
+- Create Serverless Tier and Dedicated Tier clusters.
+- Update Dedicated Tier clusters.
```
Comment (qiancai) [inline tidb-cloud/terraform-use-cluster-resource.md:15]:
```suggestion
- Modify Dedicated Tier clusters.
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 24]
```diff
@@ -0,0 +1,291 @@
...
+   +------+-------+------+
+   | id   | name  | age  |
+   +------+-------+------+
+   |    1 | Alice |   20 |
+   |    2 | Bob   |   30 |
+   +------+-------+------+
+   2 rows in set (0.24 sec)
+   ```
+
+### Create and run a S3 import task
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:182]:
```suggestion
### Create and run an AmazonS3 import task
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 25]
```diff
@@ -0,0 +1,291 @@
...
+   
+   Enter a value: yes
+   
+   tidbcloud_import.example_s3_csv: Creating...
+   tidbcloud_import.example_s3_csv: Creation complete after 3s [id=781075]
+   tidbcloud_import.example_s3_parquet: Creating...
+   tidbcloud_import.example_s3_parquet: Creation complete after 4s [id=781076]
+   ```
+   
+3. Use `terraform refresh` and `terraform state show tidbcloud_import.${resource-name}` to update and check the status just as described in the last section.
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:238]:
```suggestion
3. Use `terraform refresh` and `terraform state show tidbcloud_import.${resource-name}` to update and check the status of the import task.
```

File: tidb-cloud/terraform-use-import-resource.md [hunk 26]
```diff
@@ -0,0 +1,291 @@
...
+   | id   | name  | age  |
+   +------+-------+------+
+   |    1 | Alice |   20 |
+   |    2 | Bob   |   30 |
+   +------+-------+------+
+   2 rows in set (0.24 sec)
+   ```
+
+### Create and run a S3 import task
+
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:183]:
```suggestion
```suggestion

> **Note:**
>
> To allow TiDB Cloud to access your files in the Amazon S3 bucket, you need to [configure Amazon S3 access](/tidb-cloud/config-s3-and-gcs-access.md#configure-amazon-s3-access) first.

```
```

File: tidb-cloud/terraform-use-import-resource.md (hunk 21 above)
Comment (qiancai) [reply tidb-cloud/terraform-use-import-resource.md:291]: yes, I think so.

File: tidb-cloud/terraform-use-cluster-resource.md (hunk 23 above)
Comment (qiancai) [reply tidb-cloud/terraform-use-cluster-resource.md:15]: The Cloud console UI uses "Modify" as well.

File: tidb-cloud/terraform-use-import-resource.md [hunk 27]
```diff
@@ -0,0 +1,297 @@
...
+   2 rows in set (0.24 sec)
+   ```
+
+### Create and run an AmazonS3 import task
+```suggestion
+
+> **Note:**
+>
+> To allow TiDB Cloud to access your files in the Amazon S3 bucket, you need to [configure Amazon S3 access](/tidb-cloud/config-s3-and-gcs-access.md#configure-amazon-s3-access) first.
+```
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:188]:
```suggestion
### Create and run an Amazon S3 import task

> **Note:**
>
> To allow TiDB Cloud to access your files in the Amazon S3 bucket, you need to [configure Amazon S3 access](/tidb-cloud/config-s3-and-gcs-access.md#configure-amazon-s3-access) first.

```

File: tidb-cloud/terraform-use-import-resource.md [hunk 28]
```diff
@@ -0,0 +1,296 @@
...
+   +------+-------+------+
+   2 rows in set (0.24 sec)
+   ```
+
+### Create and run an Amazon S3 import task
+
+> **Note:**
+>
+> To allow TiDB Cloud to access your files in the Amazon S3 bucket, you need to [configure Amazon S3 access](/tidb-cloud/config-s3-and-gcs-access.md#configure-amazon-s3-access) first.
+
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:187]:
```suggestion
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

File: tidb-cloud/terraform-use-import-resource.md [hunk 29]
```diff
@@ -0,0 +1,295 @@
...
+## Prerequisites
+
+- [Get TiDB Cloud Terraform Provider](/tidb-cloud/terraform-get-tidbcloud-provider.md).
+- [Create a Serverless Tier or Dedicated Tier cluster](/tidb-cloud/create-tidb-cluster.md).
+
+You can manage either a local import task or an Amazon S3 import task using the import resource.
+
+## Create and run a local import task
+
```
Comment (qiancai) [inline tidb-cloud/terraform-use-import-resource.md:24]:
```suggestion

To create and run a local import task for a Serverless Tier cluster, take the following steps:

```

File: tidb-cloud/terraform-use-import-resource.md [hunk 30]
```diff
@@ -0,0 +1,295 @@
...
+     required_providers {
+       tidbcloud = {
+         source = "tidbcloud/tidbcloud"
+       }
+     }
+   }
+
+    provider "tidbcloud" {
+      public_key = "fake_public_key"
+      private_key = "fake_private_key"
```
Comment (ran-huang) [inline tidb-cloud/terraform-use-import-resource.md:46]: "fake" may be misleading.
```suggestion
      public_key = "your_public_key"
      private_key = "your_private_key"
```

**Review by qiancai — APPROVED** [review]
LGTM

**Review by ran-huang — APPROVED** [review]
LGTM

**Pull request merged**